classes/controller/FrontController.php

```diff
@@ -1950,4 +1946,23 @@ protected function buildContainer()
 
         return $container;
     }
+
+    /**
+     * @return array containing the URLs of the same page but for different languages
+     */
+    protected function getAlternativeLangsUrl()
+    {
+        $alternativeLangs = array();
+        $languages = Language::getLanguages(true, $this->context->shop->id);
+
+        if ($languages < 2) {
+            // No need to display alternative lang if there is only one enabled
+            return $alternativeLangs;
+        }
+
+        foreach (Language::getLanguages(true, $this->context->shop->id) as $lang) {
```

Comment:
Why not use $languages here instead of make a 2nd call to Language::getLanguages(true, $this->context->shop->id) ?